### PR TITLE
Basic Show instance of NameMap

### PR DESCRIPTION
**Pull request by fabianhjr**
This is helping me log and understand how hints change in some places, it is barebones and probably not as well formatted.

### CODE REVIEW

File: src/Data/NameMap.idr [hunk 1]
```diff
@@ -17,6 +17,11 @@ data Tree : Nat -> Type -> Type where
   Branch2 : Tree n v -> Key -> Tree n v -> Tree (S n) v
   Branch3 : Tree n v -> Key -> Tree n v -> Key -> Tree n v -> Tree (S n) v
 
+Show v => Show (Tree n v) where
+  show (Leaf k v) = "Leaf: " ++ show k ++ " -> " ++ show v ++ "\n"
+  show (Branch2 t1 k t2) = "Branch2: " ++ show t1 ++ "\n < " ++ show k ++ "\n" ++ show t2 ++ "\n"
+  show (Branch3 t1 k1 t2 k2 t3) = "Branch2: " ++ show t1 ++ "\n < " ++ show k1 ++ "\n" ++ show t2 ++ "\n < " ++ show k2 ++ "\n" ++ show t3 ++ "\n"
```
Comment (clayrat) [inline src/Data/NameMap.idr]: should be `"Branch3: "`

Comment (fabianhjr) [reply src/Data/NameMap.idr]: Oops, on it

Comment (fabianhjr) [reply src/Data/NameMap.idr]: Fixed

**Comment by edwinb** [comment]
Seems like a useful thing to have available, ta.

**Pull request merged**